### PR TITLE
1011 bleaching charts

### PR DESCRIPTION
**Pull request by mmnoo**
Changes:
- copied R code from[ Trello ticket](https://trello.com/c/Cnt8XPkw/1011-dashboard-visualizations-aggregate-translate-plotly-code-into-javascript-for-aggregate-plots-see-attached-file) to summarize bleaching data
- created bar chart for summarized bleaching data

Screenshot 2024-12-17 at 5 53 39 PM
Screenshot 2024-12-17 at 5 26 39 PM

## Summary by CodeRabbit

- **New Features**
	- Introduced new environment configuration template for application setup.
	- Added multiple issue templates for bug reports, feature requests, and general questions to enhance user communication.
	- Implemented GitHub Actions workflows for reusable deployment processes and automated deployment for different environments.
	- Enhanced README with installation instructions and mobile testing setup.
	- Added new styled components and UI elements for improved user interface and experience across various components.
	- Implemented new metrics and chart components for visualizing project data.

- **Bug Fixes**
	- Updated `.gitignore` to better manage ignored files and directories.

- **Documentation**
	- Improved README and added context in various files for better understanding and usage.

- **Chores**
	- Cleaned up and organized code structure, including the introduction of new helper functions and hooks for better state management and UI responsiveness.

### CODE REVIEW

File: src/components/MetricsPane/charts/AggregateBleaching.jsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        numColoniesTotal: numColoniesTotalForRecord + accumulator.numColoniesTotal,
+        totalColoniesNormal,
+        totalColoniesPale,
+        totalColonies0to20,
+        totalColonies20to50,
+        totalColonies50to80,
+        totalColonies80to100,
+        totalColoniesDead,
+      }
+    }, initialColoniesBleachedSummarizedBySeverity)
```
Comment (mmnoo) [inline src/components/MetricsPane/charts/AggregateBleaching.jsx:71]: @saanobhaai this is definitely worth a review by you to ensure I translated what R was doing properly. We can do a walkthrough in tomorrows meeting if that makes sense.

**Comment by mmnoo** [comment]
@julia-rom I have added test instructions to the [last PR](https://github.com/data-mermaid/mermaid-dash-v2/pull/86) in this series. Best to test it out with all the features together now that they are all wired up :)

File: src/components/MetricsPane/charts/AggregateBleaching.jsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            xbins: {
+              size: 100,
+            },
+          },
+        ]}
+        layout={{
+          ...chartTheme.layout,
+          margin: { ...chartTheme.layout.margin, b: 60 },
+          xaxis: {
+            ...chartTheme.layout.xaxis,
```
Comment (julia-rom) [inline src/components/MetricsPane/charts/AggregateBleaching.jsx:117]: could preprocess chart & layout data for better readbility, ex:

const chartData = [
  {
    y: [
      totalColoniesNormal,
      totalColoniesPale,
      totalColonies0to20,
      totalColonies20to50,
      totalColonies50to80,
      totalColonies80to100,
      totalColoniesDead,
    ],
    x: ['Normal', 'Pale', '0-20%', '20-50%', '50-80%', '80-100%', 'Dead'],
    type: 'bar',
    marker: {
      color: chartTheme.aggregateCharts.bleaching.marker.color,
    },
  },
];

const chartLayout = {
  ...chartTheme.layout,
  margin: { ...chartTheme.layout.margin, b: 60 },
  xaxis: {
    ...chartTheme.layout.xaxis,
    title: {
      ...chartTheme.layout.xaxis.title,
      text: 'Bleaching severity',
    },
  },
  yaxis: {
    ...chartTheme.layout.yaxis,
    title: {
      ...chartTheme.layout.yaxis.title,
      text: 'Number of coral colonies',
    },
  },
}

**Review by julia-rom — APPROVED** [review]
look good, made a small suggestion

**Pull request merged**